under-review/sdfobject-operational_state.sdf.json

```diff
@@ -0,0 +1,56 @@
+{
+  "info": {
+    "copyright": "copyright 2016-2017, 2019 Open Connectivity Foundation, Inc. All rights reserved.",
+    "license": "https://github.com/one-data-model/oneDM/blob/master/LICENSE",
+    "title": "Operational State",
+    "version": "2019-02-15"
+  },
+  "sdfObject": {
+    "operational.state": {
+      "description": "This Resource describes the operational and job states on a device. The states can be read or set, setting indicates a desired state. A device may reject an attempt to set a state that would result in adverse operational characteristics. The Property \"machineStates\" is an array of the possible operational states. The Property \"currentMachineState\" is the current state of operation of the device. The Property \"jobStates\" is an array of the possible job states. The Property \"currentJobState\" is the currently active jobState. The Property \"runningTime\" is the ISO8601 encoded elapsed time in the current operational state. The Property \"remainingTime\" is the ISO8601 encoded time till completion of the current operational state. The Property \"progressPercentage\" is the percentage completeness of the current jobState.",
+      "sdfProperty": {
+        "currentMachineState": {
+          "description": "The current state of operation of the device.",
+          "type": "string"
+        },
+        "currentJobState": {
+          "description": "The currently active jobState.",
+          "type": "string"
+        },
+        "machineStates": {
+          "description": "The array of the possible operational states.",
+          "items": {
+            "type": "string"
+          },
+          "writable": false,
+          "type": "array"
+        },
+        "runningTime": {
+          "description": "A string representing duration formatted as defined in ISO 8601. Allowable formats are: P[n]Y[n]M[n]DT[n]H[n]M[n]S, P[n]W, P[n]Y[n]-M[n]-DT[0-23]H[0-59]:M[0-59]:S, and P[n]W, P[n]Y[n]M[n]DT[0-23]H[0-59]M[0-59]S. P is mandatory, all other elements are optional, time elements must follow a T.",
+          "type": "string",
+          "pattern": "^(P(?!$)([0-9]+Y)?([0-9]+M)?([0-9]+W)?([0-9]+D)?((T(?=[0-9]+[HMS])([0-9]+H)?([0-9]+M)?([0-9]+S)?)?))$|^(P[0-9]+W)$|^(P[0-9]{4})-(1[0-2]|0[1-9])-(3[0-1]|2[0-9]|1[0-9]|0[1-9])T(2[0-3]|1[0-9]|0[1-9]):([0-5][0-9]):([0-5][0-9])$|^(P[0-9]{4})(1[0-2]|0[1-9])(3[0-1]|2[0-9]|1[0-9]|0[1-9])T(2[0-3]|1[0-9]|0[1-9])([0-5][0-9])([0-5][0-9])$"
```

Comment:
We have time type we should use here so that each ecosystem can use the proper way of encoding in that ecosystem

Reply:
we still need a way to define time.
we can isolate time as part of the infrastructure, e.g. something that gets referenced.
however we should only do that when another object is brought forward that uses the same element

Reply:
sdfobject-door.sdf.json openDuration is another ISO 8601 duration.

Reply:
I guess that time and duration definitions should be aligned, if unix-time is used for time then duration is the difference between two unix-times, and vice versa if ISO8601 time is used then ISO8601 duration is used